common/oe_host_socket.h

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifndef _OE_HOST_SOCKET_H
+#define _OE_HOST_SOCKET_H
+
+#include <openenclave/corelibc/bits/types.h>
```

Comment:
Headers need to be wrapped within a `OE_EXTERNC_BEGIN .. OE_EXTERNC_END` pair so that the header can be inculded from C++ as well.